iam/policy/statement.go

```diff
@@ -127,8 +127,14 @@ func (statement Statement) isValid() error {
 	}
 
 	for action := range statement.Actions {
-		if !statement.Resources.objectResourceExists() && !statement.Resources.bucketResourceExists() {
-			return Errorf("unsupported Resource found %v for action %v", statement.Resources, action)
+		if action.isObjectAction() {
+			if !statement.Resources.objectResourceExists() {
+				return Errorf("unsupported Resource found %v for action %v", statement.Resources, action)
+			}
+		} else {
+			if !statement.Resources.bucketResourceExists() {
```

Comment:
I am not sure - we have some admin:* actions as well, and I don't think they have bucket resources.. wouldn't that fail in that case @donatello ? (basically adding a IAM policy which contains admin:* actions

Reply:
This is statement level validation - for admin requests it is validated earlier in line 95, using a different logic.